### PR TITLE
LP-2371 Implement and Integrate Application review

### PR DESCRIPTION
**Pull request by asadali145**
[LP-2371](https://philanthropyu.atlassian.net/browse/LP-2371)

[Theme PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/429)

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2574/workflows/workflows_id/jobs/2574/artifacts)**. Coverage !!! :scream:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/serializers.py (72.7%): Missing lines 29-31
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (62.5%): Missing lines 37-40,42,49,58-60
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (33.3%): Missing lines 121-124,137,151-155,157-160,162-163
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (33.3%): Missing lines 84-85,87,102-103,112,126,128-130,132-133,137-138,140,155,169-171,173-175,177-178,180-181
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (63.6%): Missing lines 309,313-314,319
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   128 lines
Missing: 58 lines
Coverage: 54%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2627/workflows/workflows_id/jobs/2627/artifacts)**. Coverage !!! :scream:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (54.8%): Missing lines 189-193,195-198,200-201,233-234,237
openedx/adg/admin_dashboard/applications/mixins.py (94.1%): Missing lines 43,61
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (28.4%): Missing lines 179,181,185-190,196-197,199,213-215,217-219,221-222,224-225,234,240-241,243,255,257-259,261-262,266-268,270,289,302,304-306,308,310-312,314,316-317,319-320,322,331-332,340
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (63.6%): Missing lines 363,367-368,373
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   283 lines
Missing: 73 lines
Coverage: 74%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2642/workflows/workflows_id/jobs/2642/artifacts)**. Coverage !!! :scream:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (54.8%): Missing lines 189-193,195-198,200-201,233-234,237
openedx/adg/admin_dashboard/applications/mixins.py (94.9%): Missing lines 45,63
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (29.6%): Missing lines 182,184,188-193,199-200,202,216-218,220-222,224-225,227-228,237,243-244,246,258,260-262,264-265,269-272,274,323,325-327,329,331-333,335,337-338,340-341,343
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (63.6%): Missing lines 363,367-368,373
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   290 lines
Missing: 70 lines
Coverage: 75%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2668/workflows/workflows_id/jobs/2668/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (100%)
openedx/adg/admin_dashboard/applications/mixins.py (94.9%): Missing lines 45,63
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (94.4%): Missing lines 236,242-243,245
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   390 lines
Missing: 6 lines
Coverage: 98%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2675/workflows/workflows_id/jobs/2675/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (100%)
openedx/adg/admin_dashboard/applications/mixins.py (94.9%): Missing lines 45,63
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (94.4%): Missing lines 237,243-244,246
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   390 lines
Missing: 6 lines
Coverage: 98%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2686/workflows/workflows_id/jobs/2686/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (100%)
openedx/adg/admin_dashboard/applications/mixins.py (95.1%): Missing lines 51,69
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (94.5%): Missing lines 240,246-247,249
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
openedx/adg/lms/applications/tests/test_helpers.py (100%)
openedx/adg/lms/applications/tests/test_models.py (100%)
-------------
Total:   425 lines
Missing: 6 lines
Coverage: 98%
-------------
```

File: openedx/adg/admin_dashboard/applications/views.py [hunk 1]
```diff
@@ -202,6 +154,188 @@ def get_context_data(self, *, object_list=None, **kwargs):
...
+            'next_application_url': next_application_url,
+            'save_note_url': reverse('save-note', kwargs={'pk': application.id}),
+
+            'accept_button_class': 'green' if is_bu_pre_reqs_passed else DISABLED_BUTTON_CLASS,
+            'waitlist_button_class': 'light-orange' if is_bu_pre_reqs_passed else DISABLED_BUTTON_CLASS,
+            'finish_and_send_button_class': 'aqua-blue' if is_bu_pre_reqs_passed else DISABLED_BUTTON_CLASS,
+
+            'next_btn_class': (
+                'dark-blue'
+                if next_application_url and next_application_url != 'None' else DISABLED_BUTTON_CLASS
```
Comment (annas-usman) [inline openedx/adg/admin_dashboard/applications/views.py:301]: Maybe for better readability we can break it as:
```
(
    'dark-blue'
    if next_application_url and next_application_url != 'None' 
    else DISABLED_BUTTON_CLASS
)

File: openedx/adg/admin_dashboard/applications/views.py [hunk 2]
```diff
@@ -202,6 +154,188 @@ def get_context_data(self, *, object_list=None, **kwargs):
...
+
+        if self.request.method == 'POST':
+            next_application_url = self.request.POST.get('next_application_url')
+            prev_application_url = self.request.POST.get('previous_application_url')
+
+            return prev_application_url, next_application_url
+
+        curr_id_idx = self.application_ids.index(self.object.id)
+        previous_id = self.application_ids[curr_id_idx - 1] if curr_id_idx > 0 else None
+        next_id = self.application_ids[curr_id_idx + 1] if curr_id_idx < len(self.application_ids) - 1 else None
```
Comment (annas-usman) [inline openedx/adg/admin_dashboard/applications/views.py:332]: Maybe we should break these lines for readability

File: openedx/adg/lms/applications/models.py [hunk 3]
```diff
@@ -354,6 +357,30 @@ def set_status(self, status):
         self.status = status
         self.save()
 
+    def send_application_status_update_email(self, message_for_applicant):
+        """
```
Comment (annas-usman) [inline openedx/adg/lms/applications/models.py:361]: shouldn't we write this function in helpers instead of models?

**Review by annas-usman — CHANGES_REQUESTED** [review]

File: openedx/adg/admin_dashboard/applications/helpers.py [hunk 4]
```diff
@@ -142,3 +148,94 @@ def get_date_and_course_status_ordering(application_type, order_by):
...
+    Args:
+        user (User): Admin user
+        application (UserApplication): User application
+
+    Returns:
+        bool: True if admin has access else False
+    """
+    return (
+        user.is_superuser or
+        is_adg_admin(user) or
```
[the remark this answers is not in the document]
Comment (asadali145) [reply openedx/adg/admin_dashboard/applications/helpers.py:221]: Nice one!

File: openedx/adg/admin_dashboard/applications/tests/test_helpers.py [hunk 5]
```diff
@@ -170,3 +177,175 @@ def test_get_date_and_course_status_ordering(application_type, order_by, expecte
...
+        ('adg_admin', True),
+        ('bu_admin', True),
+    ]
+)
+@skip_unless_lms
+@pytest.mark.django_db
+def test_user_has_no_access_to_application(user_with_profile, user_type, expected_result):
+    """
+    Tests `has_access_to_application` returns expected result for different type of users.
+    """
```
[the remark this answers is not in the document]
Comment (asadali145) [reply openedx/adg/admin_dashboard/applications/tests/test_helpers.py:196]: My bad, Duplicated it. Written on 2 different days :)

File: openedx/adg/admin_dashboard/applications/views.py [hunk 6]
```diff
@@ -202,6 +154,188 @@ def get_context_data(self, *, object_list=None, **kwargs):
...
+        applications = UserApplication.objects.submitted_applications().filter(
+            status=self.application_type
+        ).select_related(
+            'user', 'user__profile', 'user__extended_profile', 'business_line', 'user__application_hub'
+        )
+
+        filtered_applications = self.get_filtered_applications(applications)
+        ordering = get_date_and_course_status_ordering(self.application_type, self.order_by)
+
+        return filtered_applications.order_by(*ordering)
```
[the remark this answers is not in the document]
Comment (asadali145) [reply openedx/adg/admin_dashboard/applications/views.py:249]: This is the bare minimum duplication that we can go with, Me and @aightmunam had discussions around it to create common methods and we did that and moved them to mixin but this is what we should not move to common as passing params to get_queryset doesn't make sense.

If you have some other ideas then LMK.

File: openedx/adg/lms/applications/models.py (hunk 3 above)
Comment (asadali145) [reply openedx/adg/lms/applications/models.py:361]: Me and @harris-irfan had a discussion on this one and we were a bit confused in the 2 implementations and If we think like it is related to the application object then it makes some sense but in other case it is not performing some action over the object. Moving it to helpers as you were not part of discussion and you pointed it out.

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2689/workflows/workflows_id/jobs/2689/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (100%)
openedx/adg/admin_dashboard/applications/mixins.py (95.1%): Missing lines 51,69
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (94.6%): Missing lines 241,247-248,250
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   378 lines
Missing: 6 lines
Coverage: 98%
-------------
```

File: openedx/adg/admin_dashboard/applications/tests/test_helpers.py [hunk 7]
```diff
@@ -170,3 +177,146 @@ def test_get_date_and_course_status_ordering(application_type, order_by, expecte
...
+
+
+@pytest.mark.parametrize(
+    'date_str, expected_result',
+    [
+        ('1/01/20', 'January 01st, 2020'),
+        ('2/01/20', 'January 02nd, 2020'),
+        ('3/01/20', 'January 03rd, 2020'),
+        ('4/01/20', 'January 04th, 2020'),
+        ('5/01/20', 'January 05th, 2020'),
```
Comment (harris-irfan) [inline openedx/adg/admin_dashboard/applications/tests/test_helpers.py:189]: This comment is unrelated to the tests, and more related to the user facing text. Would it make sense to show a zero appended at the start for days less than 10, like 01st, 02nd, etc.? Can we somehow alter the original function to remove these zeros?

File: openedx/adg/admin_dashboard/applications/views.py [hunk 8]
```diff
@@ -197,6 +151,188 @@ def get_context_data(self, *, object_list=None, **kwargs):
...
+            **kwargs (dict): Keyword Arguments
+
+        Returns:
+            Response: TemplateResponse
+        """
+        application = self.object
+        msg_for_applicant = request.POST.get('notes-for-applicant', '')
+        decision = request.POST.get('decision', application.status)
+
+        if decision != application.status:
```
[the remark this answers is not in the document]
Comment (asadali145) [reply openedx/adg/admin_dashboard/applications/views.py:224]: It is just a backend validation, That is restricted using JS and user can alter that from frontend. Just to be on the safe side.

File: openedx/adg/admin_dashboard/applications/views.py [hunk 9]
```diff
@@ -197,6 +151,188 @@ def get_context_data(self, *, object_list=None, **kwargs):
...
+        bu_pre_reqs_info = get_course_info_for_application_review(bu_pre_req_courses, user)
+        previous_application_url, next_application_url = self.previous_and_next_application_url()
+        is_bu_pre_reqs_passed = application_hub.bu_prereq_courses_status == ApplicationHub.COMPLETED
+
+        context.update({
+            'user': application.user,
+            'user_profile': user.profile,
+            'extended_profile': user.extended_profile,
+            'work_experiences': application.applications_workexperiences.all(),
+            'educations': application.applications_educations.all(),
```
[the remark this answers is not in the document]
Comment (asadali145) [reply openedx/adg/admin_dashboard/applications/views.py:283]: I think this is quite big enough story and we should avoid any further changes which will in the end impact QA and also the reviewers, We can surely do this under some smaller story or we can create a story for such improvements project wide.

File: openedx/adg/admin_dashboard/applications/tests/test_helpers.py (hunk 7 above)
Comment (asadali145) [reply openedx/adg/admin_dashboard/applications/tests/test_helpers.py:189]: If we add `-` before `%d` in the date format it will remove the prepended `0`.

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2695/workflows/workflows_id/jobs/2695/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (100%)
openedx/adg/admin_dashboard/applications/mixins.py (95.1%): Missing lines 51,69
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (94.6%): Missing lines 241,247-248,250
openedx/adg/common/helpers.py (100%)
openedx/adg/common/tests/test_helpers.py (100%)
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   374 lines
Missing: 6 lines
Coverage: 98%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2696/workflows/workflows_id/jobs/2696/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (100%)
openedx/adg/admin_dashboard/applications/mixins.py (95.7%): Missing lines 52,86
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (97.2%): Missing lines 234,237
openedx/adg/common/helpers.py (100%)
openedx/adg/common/tests/test_helpers.py (100%)
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   378 lines
Missing: 4 lines
Coverage: 98%
-------------
```

**Review by annas-usman — APPROVED** [review]
Great Work!

File: openedx/adg/admin_dashboard/applications/mixins.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    filter_by_business_line = ''
+    omni_courses_query_values = []
+    bu_courses_query_values = []
+    business_line_query = []
+
+    def setup(self, request, *args, **kwargs):
+        """
+        Add query params as attributes to the class
+        """
+        super().setup(request, *args, **kwargs)
```
[the remark this answers is not in the document]
Comment (harris-irfan) [reply openedx/adg/admin_dashboard/applications/mixins.py:36]: But each of the setup methods of those views have their own super calls, e.g.:

```
class ApplicationDetailView(EnsureAdminUserAuthenticatedMixin, FilterApplicationsMixin, DetailView, View):
    """
    View for application review.
    """

    template_name = 'adg/admin_dashboard/applications/application_review.html'
    http_method_names = ('get', 'post')
    object = None
    queryset = None
    application_ids = []
    context_object_name = 'application'

    def setup(self, request, *args, **kwargs):
        """
        Set class attributes.
        Args:
            request (request): Request object
            *args (list): Arguments
            **kwargs (dict): Keyword Arguments
        """
        super().setup(request, *args, **kwargs)
```

File: openedx/adg/admin_dashboard/applications/api/tests/test_views.py [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+from openedx.adg.admin_dashboard.applications.helpers import get_formatted_date_for_application_review_notes
+from openedx.adg.lms.applications.constants import ADG_ADMIN_GROUP_NAME
+from openedx.adg.lms.applications.tests.factories import BusinessLineFactory, UserApplicationFactory
+from openedx.core.djangolib.testing.utils import skip_unless_lms
+
+
+@pytest.fixture(name='adg_admin_user_client')
+def adg_admin_user_client_login(user_client):
+    """
+    Superuser and client login fixture.
```
Comment (harris-irfan) [inline openedx/adg/admin_dashboard/applications/api/tests/test_views.py:21]: Docstring to be updated, no longer a superuser.

File: openedx/adg/admin_dashboard/applications/mixins.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (asadali145) [reply openedx/adg/admin_dashboard/applications/mixins.py:36]: The super call you mentioned is to call setup of mixin and mixin's will call the setup of detail and list generic views.

File: openedx/adg/admin_dashboard/applications/api/tests/test_views.py (hunk 11 above)
Comment (harris-irfan) [reply openedx/adg/admin_dashboard/applications/api/tests/test_views.py:21]: @asadali145 this remains unaddressed

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/2697/workflows/workflows_id/jobs/2697/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/admin_dashboard/applications/api/permissions.py (100%)
openedx/adg/admin_dashboard/applications/api/serializers.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/constants.py (100%)
openedx/adg/admin_dashboard/applications/api/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/api/urls.py (100%)
openedx/adg/admin_dashboard/applications/api/views.py (100%)
openedx/adg/admin_dashboard/applications/constants.py (100%)
openedx/adg/admin_dashboard/applications/helpers.py (100%)
openedx/adg/admin_dashboard/applications/mixins.py (95.7%): Missing lines 52,86
openedx/adg/admin_dashboard/applications/tests/test_helpers.py (100%)
openedx/adg/admin_dashboard/applications/tests/test_views.py (100%)
openedx/adg/admin_dashboard/applications/urls.py (100%)
openedx/adg/admin_dashboard/applications/views.py (97.2%): Missing lines 234,237
openedx/adg/common/helpers.py (100%)
openedx/adg/common/tests/test_helpers.py (100%)
openedx/adg/conftest.py (100%)
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/models.py (100%)
openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py (100%)
-------------
Total:   378 lines
Missing: 4 lines
Coverage: 98%
-------------
```
